src/Microsoft.Identity.Client/UI/CustomWebUiHandler.cs

```diff
@@ -81,26 +83,30 @@ public CustomWebUiHandler(ICustomWebUi customWebUi)
                     };
                 }
 
-                throw new MsalCustomWebUiFailedException(CoreErrorMessages.CustomWebUiRedirectUriWasNotMatchedToProperUri(uri.AbsolutePath, redirectUri.AbsolutePath));
+                throw new MsalCustomWebUiFailedException(
+                    CoreErrorMessages.CustomWebUiRedirectUriWasNotMatchedToProperUri(
+                        uri.AbsolutePath,
+                        redirectUri.AbsolutePath));
             }
-            catch (Exception ex)
+            catch (OperationCanceledException)
             {
-                var authStatus = AuthorizationStatus.UnknownError;
 
-                if (ex is OperationCanceledException)
-                {
-                    requestContext.Logger.Info(LogMessages.CustomWebUiOperationCancelled);
-                    authStatus = AuthorizationStatus.UserCancel;
-                }
-                else
+                requestContext.Logger.Info(LogMessages.CustomWebUiOperationCancelled);
+                return new AuthorizationResult(AuthorizationStatus.UserCancel, null);
+            }
+            catch (Exception ex)
```

Comment:
This is a little fix I found while implemeting a custom web ui - if the implementation throws an unexpected exception, then encapsualte it into MsalCustomWebUiFailedException, which is still an MsalException, but has more meaning than the exissting "UnkownException".